src/components/reusables/Form.tsx

```diff
@@ -30,20 +32,16 @@ export default class Form extends React.Component<FormProps, {}> {
     event.preventDefault();
     let form = (this.refs["form"] as any);
```

Comment:
But, to go with the `messageRef`, creating a ref for this form is more complicated because its ref is referenced by parent components.

Reply:
Could you explain this more? I don't see why it's more complicated.

Reply:
The ref for the form is just as simple to create as the `messageRef` but that ref gets complicated because it's called by parent components. `SearchForm` imports `Form` and uses the `Form`'s ref to reset it: `(this.refs["form-component"] as any).refs["form"].reset();`. Right now this works but setting it up with the new Ref API doesn't work unless more ref forwarding is added. I tried it and got close to making it work but there were some issues I couldn't easily fix and it was also out of scope. The syntax and extra boilerplate was also tedious to work with.

More of ref forwarding here: https://reactjs.org/docs/forwarding-refs.html

Reply:
At some point in the not-too-distant future, I'll be working on making `Form` more easily reusable; I'll make a note to take a look at this then!  (Off the top of my head: `Form` should probably have its own built-in `clear` method anyway, which will get rid of this issue altogether.)

Reply:
Yeah, it makes sense to have the `clear` function in the form itself. I can make a different PR for the functional forwarding ref way of doing it but not sure if it will be helpful or not. I'll just wait for your implementation and see.